pkg/envoy/server.go

```diff
@@ -1532,7 +1539,7 @@ func getDirectionNetworkPolicy(ep logger.EndpointUpdater, l4Policy policy.L4Poli
 					if !cs {
 						canShortCircuit = false
 					}
-					if len(rule.RemotePolicies) == 0 && rule.L7 == nil && rule.DownstreamTlsContext == nil && rule.UpstreamTlsContext == nil {
+					if len(rule.RemotePolicies) == 0 && rule.L7 == nil && rule.DownstreamTlsContext == nil && rule.UpstreamTlsContext == nil && rule.ServerNames == nil {
```

Comment:
I wonder if `len(rule.ServerNames) == 0` would not be a better check here, but I think that the way `ServerNames` is initialized makes it impossible to happen.

AFAIU, there is no way for `ServerNames` to have it's length decreased to 0, right?

Reply:
You are right, checking len for 0 is more robust test, as it would keep working also when the slice is initialized but has no entries. Will change :-)